[FIO toup] watchdog: fix imx7ulp init

ldts opened the pull request:
These fixes came from Codeaurora - basically they enforce the
reconfiguration sequence timing requirements. We should have them
upstream (I need to find the original author to credit)

Signed-off-by: Jorge Ramirez <jorge@foundries.io>

Please do not submit a Pull Request via github.  Our project makes use of
mailing lists for patch submission and review.  For more details please
see https://www.denx.de/wiki/U-Boot/Patches

Commit:
[FIO nxp] MLK-22654 mx7ulp: Update wdog disable sequence

Update the mx7ulp wdog disable sequence to avoid potential reset issue
in unlock or refresh sequence. Both sequence need two words write
to wdog CNT register in 16 bus clocks window, if miss the window,
the write will cause violation in wdog and reset the chip.

Current u-boot code is using writel() function which has a DMB barrier
to order the memory access. The DMB between two words write may introduce
some delay in certain circumstance, causing the wdog reset due to 16 bus
clock window requirement.

This patch replaces writel() function by __raw_writel() to avoid such issue,
and improve to check if watchdog is already disabled or unlocked.

Signed-off-by: Ye Li <ye.li@nxp.com>
Tested-by: Breno Lima <breno.lima@nxp.com>
Reviewed-by: Peng Fan <peng.fan@nxp.com>
(cherry picked from commit b8c99d5f5bcc5573d3394b68890db16b6bb5fc88)

ricardosalveti commented on arch/arm/mach-imx/mx7ulp/soc.c:
```diff
@@ -104,14 +104,28 @@ int board_postclk_init(void)
-	writel(REFRESH_WORD1, (wdog_base + 0x04));
+	u32 val_cs = readl(wdog_base + 0x00);
+
+        if (!(val_cs & 0x80))
```
Seems indentation is wrong here, probably tab x spaces.

ricardosalveti commented:
Looks like this one is based on https://source.codeaurora.org/external/imx/uboot-imx/commit/?id=b26fb3c9b0e76075eada24a087d2db5f6fd3fc73? If so it might be better to cherry-pick or at least reference as part of your commit.

ldts committed:
[FIO extras] mx7ulp: update wdog disable sequence

WDOG1 might have been enabled already depending on FUSE hence we need
to be as close as possible to its reconfiguration timing requirement
of 128 bus clock limit.

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>

ldts commented:
done

ricardosalveti reviewed (approved):
LGTM
